### PR TITLE
Updates to the contribution guidelines

### PR DESCRIPTION
**Pull request by mavam**
This PR contains updates regarding unit testing and doxygen comments.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -356,3 +357,77 @@ When integrating 3rd-party code into the code base, use the following scaffold:
...
+
+  #define SUITE foo
+  #include "test.hpp"
+  #include "fixtures/my_fixture.hpp"
+
+  #include <iostream>     // standard library includes
+
+  #include <caf/...>      // CAF includes
+
+  #include "vast/foo.hpp" // VAST includes
```
Comment (Neverlord) [inline CONTRIBUTING.md]: This is the unit-under-test. Why did you opt against including this first?

Comment (mavam) [reply CONTRIBUTING.md]: I think this was an oversight. I wanted to write `#include "vast/..."` but my mental context was still within `foo`. Will fix.

**Comment by mavam** [comment]
FYI: I've rebased the commits to make Jenkins happy.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Neverlord) [reply CONTRIBUTING.md]: I think this include should come first in source fies.

**Comment by Neverlord** [comment]
While we're at it:

```
- Do not use the inline keyword unless to avoid duplicate symbols.
```

We had this discussion a while ago. I'm still in favor of inlining trivial functions like getters that simply return a member.

```
- All header files should use #define guards to prevent multiple inclusion.
```

We switched to `#pragma once` recently.

```
- Break function arguments after the comma ...
```

Both examples to this rule still use `T const&`.

```
- For multi-line comments, break after the
```

The line under `@param x` misses `///`.

**Comment by mavam** [comment]
> I'm still in favor of inlining trivial functions like getters that simply return a member.

I don't have particularly strong opinions. Should we bound the number of lines for what "trivial" means? Or apply the Rule of Common Sense?

**Comment by Neverlord** [comment]
> Should we bound the number of lines for what "trivial" means?

I think a good rule of thumb is 1-3 lines.

**Comment by mavam** [comment]
I've updated the guidelines accordingly.

**Pull request merged**